### PR TITLE
Ports VG's Plasmamen + Species breathing additions

### PR DESCRIPTION
**Pull request by RemieRichards**
## Changes:
- Adds Plasmamen, a species that breathes plasma, and sets on fire if they don't wear a specific class of suit
- Adds Plasmamen suits, which can extinguish a plamaman wearing it, a limited amount of times.
- Allows Species to breathe different gases, and to also have too much Oxygen, etc. (Only species using these effects at the moment is Plasmamen. Humans and other species are NOT affected)
- Adds Plasmamen breath tanks
- Adds Plasmamen equipment to select equipment verb
- Adds Alerts for having too much plasma, too little CO2, too much oxygen, etc.
- Extracted a piece of breathe code out to be a `/datum/gas_mixture` proc.
- Produced a proc to do the reverse of the above proc.
- Extracted out the code that runs when there's too little oxygen, generalised it, and used it for all breathable gases, to make all breathable gases functionally equivalent.
- Adds ability for species to override drawing fire on the mob (maybe the mob is fire immune or maybe it draws it's own unique fire like plasmamen)
- Lack of a breathable gas is now a black alert themed around the gas, while too much of a breathable gas is now a red alert themed around the gas( Eg. no oxygen (red) --> no oxygen (black)) - Thanks @WJohn 
- Skeletons resprited to match new Plasmamen skelebones - Thanks @WJohn 
- New on fire overlay for all Humanoid mobs, to replace the current "Standing" icon_state
## Stretch goals:
- [x] Replace plasmaman underlay with a version edited to be an OVERlay - Requires spriter.
## Disclaimer:

They do NOT spawn on station, they are an admin only thing at the moment.
Also, Fully tested:
- Humans breathe the same
- Plasmamen breathe plasma
- Plasmamen suits extinguish
- Minor things, etc.
## Image reference:

plasmamen

### CODE REVIEW

**Comment by GunHog** [comment]
Pictures pl0x?

File: code/_onclick/hud/alert.dm [hunk 1]
```diff
@@ -87,15 +87,35 @@
...
+	name = "Choking (No CO2)"
+	desc = "You're not getting enough carbon dioxide. Find some good air before you pass out!"
+
+/obj/screen/alert/too_much_co2
+	name = "Chocking (CO2)"
+	desc = "There's too much carbon dioxide in the air, and you're breathing it in! Find some good air before you pass out!"
+
+/obj/screen/alert/not_enough_tox
+	name = "Choking (No Plasma)"
+	desc = "You're not getting enough plama. Find some good air before you pass out!"
```
Comment (Niknakflak) [inline code/_onclick/hud/alert.dm]: Plasma* instead of plama

**Comment by Thunder12345** [comment]
We skelebones now

**Comment by WJohn** [comment]
Perhaps for consistency's sake we should have it where lacking oxygen or co2 or plasma it should be black, where as having too much it should be red, so it fits in with pressure warnings.

**Comment by RemieRichards** [comment]
@WJohn you can art overhaul the sprites for the alerts at any time, at the moment, they work and don't look poopy, that's fine.

**Comment by WJohn** [comment]
Done already, it'd be a piece of cake to put these in in this PR:

http://www.mediafire.com/download/d4jf9w8g2tcifni/SKELETONS.rar

**Comment by RemieRichards** [comment]
@WJohn Implemented, thankies.

**Comment by Fox-McCloud** [comment]
maybe one day...maybe....


one can hope...

**Comment by Supermichael777** [comment]
Wait are these skeletons that need plasma to breathe and set plasma on fire when not in a suit. they probably need to get reduced fire damage at some level or i don't see them lasting long.

**Comment by RemieRichards** [comment]
@Supermichael777 the way they work on VG, they spawn in job specific versions of their suit (for a future PR) which is usually the only suit they'll ever need.

**Comment by pudl** [comment]
stupid idea and stupid excuse to make skelebones a meme

**Comment by WJohn** [comment]
Stop hatin' on everything.

**Comment by Kromgar** [comment]
Add the possibility to find their suits on the asteroid in an artifact room. Also can you turn into one using a green slime? If so how would you survive

**Comment by WJohn** [comment]
Also, I'm slightly curious. How does this work with oxygen tanks like on mining or what you may get out of an O2 locker? Would they all become dangerous?

**Comment by RemieRichards** [comment]
Oxygen is not innately harmful to Plasmamen, they just need to breath plasma instead.
As I said, they're not on station at the moment, which is because of these potential issues.

But they'll make fun spookies for now.

**Comment by WJohn** [comment]
I meant for humans, does too much oxygen hurt them? How does this affect oxy tanks found everywhere?

**Comment by Lumipharon** [comment]
I think he means where you say 'they can have to much oxygen' then list humans as one of the species effected by this.

So will pure oxygen negatively effect people or what?

**Comment by RemieRichards** [comment]
Oh No no, if you read the PR OP, having "too much" oxygen, is just because oxygen didn't have a "too much" where as plasma and co2 had "too much" but not a "too little"

it's just for customisability, and was trivial to implement.

@Lumipharon, She.
Also if you get me an amount, in kPA of oxygen that SHOULD harm people, I can add that.
But for now, Oxygen is only harmful if you don't have enough (should your species breathe oxygen)

**Comment by GoonOnMoon** [comment]
This is great.

**Comment by Lumipharon** [comment]
According to wikipedia, oxygen toxicity can have effects at 200-300kpa of pure oxygen. After like, 3 hours of exposure.

I wouldn't worry about, just needless complication.

**Comment by Incoming5643** [comment]
No such thing as admin only when it comes to mutantraces, as the staff of change will still have a slight chance to pick it. It be an intensely unlucky roll to experience, under a 1% chance, but then that's just how magic is.

The magic mirror could also technically be used to go plasmaman, but that seems like the kind of mistake a person would only make once.

The new unstable mutantion toxin will probably need to blacklist these races.

Also we have exotic blood, make em  bleed plasma!

**Comment by WJohn** [comment]
That's a great idea, actually. Though at the same time they're skeletons, so there's not exactly much blood I could imagine in em.

**Comment by pudl** [comment]
g i m m i c k

**Comment by SconesC** [comment]
Does this mean we got vox someday

**Comment by RemieRichards** [comment]
@Lumipharon Oki doki, pointless then.

@Incoming5643 Again, it's admin only, or a rare punishment, without the suit nobody in their right mind should pick burny skeleton man.

Also as to exotic blood @WJohn sums it up with "They're skeletons" 

Coming when I get home:
* Healing from plasma (reagent)
* Editing the Plasmaman fire underlay to work as an overlay (by cropping out the parts where a skeleman would be in the overlay) since the code gets MUCH cleaner that way.
* Rebalance of dieing from not breathing other gases (It's not really close to how not breathing oxygen is, but I can fix that)

**Comment by Lumipharon** [comment]
I'd like to imagine cultists or something summoning screaming armies of plasma skellingtons to set the crew on fire as they streak down the halls.

**Comment by RemieRichards** [comment]
@WJohn I've got a long boring task that I couldn't complete myself to do with sprites:
At the moment, the Plasmamen fire sprite is an UNDERlay, to make the code nicer and cleaner, I'd like it to be an overlay... which involves overlaying the plasmaman sprite over the fire, and cropping out the man,and then doing any extra tweaks, since the fire will go over EVERYTHING to do with the man, not just the naked version.

so if you'd please :3

**Comment by RemieRichards** [comment]
## CHANGES:
* Not having enough [insert_your_breathable_gas_here] is now identical to if you didn't have enough oxygen, as opposed to being OP, (the code for the lack of oxygen was generalised for this task)
* Breathed carbon dioxide comes out as oxygen (think plants), Breathed plasma comes out as oxygen (Idk, made it up) and as always, Breathed oxygen comes out as carbon dioxide.
* Plasmamen can heal burn and brute damage by using the plasma reagent (Injected, Breathed in, etc.)

**Comment by pudl** [comment]
bad idea

**Comment by RemieRichards** [comment]
Well ok then you've officially invalidated your feedback.
Once? Yeah sure, Ok I get it, you hate some things
Twice? ooook
Thrice? Nope, nada, ignoring you.

**Comment by WJohn** [comment]
http://www.mediafire.com/download/d4jf9w8g2tcifni/SKELETONS.rar

Here you go.

Honestly now that I look at it, I feel like this should just replace the regular fire overlay sprite I made like 3ish years ago, it's hideously outdated.

**Comment by pudl** [comment]
you wanna know why its bad?
first off: what's the point in adding a new species? a lot of people have gotten upset over lizardpeople being a thing, and here you are adding a new species. (the normal breathing is fine, plasmamen aren't)
next, fluffwise, how do these creatures exist? why would they manifest as skeletons instead of a less ridiculous creature? how on god's green earth does any organic life form live without some supply of oxygen?
even worse, skeletons were nerfed to hell because they were literally invincible to 90% of everything, and here these plasmamen are just as op. antag plasmaman? flood station with plasma, if anyone catches you just unequip your suit and set everything on fire.
and if this remains admin-only, what's the point in such a complicated race that is rarely used? the cool factor is no excuse for such a massive addition.
really, this creature is just a gimmick that doesn't need to be added. it's dumb.

**Comment by RemieRichards** [comment]
@pudl 
* people got butthurt over lizardpeople because they don't like animal erp, (read: furry culture), Skeletons aren't an animal, and the same if not less sex appeal than humans (why did i have to say this, you made me say this)
* fluffwise: We have wizards, Not to mention the Sun of the SS13 solar system nor the plasma gas giant have been explored, Those two elements could come together form a coherent backstory
* Skeletons were never meant to be a normal playable race, not to mention they were OP for only a fraction of their existence (Before the species rewrite they functioned pretty similairly to humans, if not identically)
* Evolution is stupid, present animals in SS13 already show these traits: Space carp, they don't even breath, given the 0 gases in the vacuum of space, Not to mention nowhere has it been decreed that they evolved on earth. We evolved to breath oxygen because it was a suitable gas that was in a good enough supply, for plasmamen (who at this point probably do hail from the plasma gas giant) most likely only had plasma to go on, for whatever reason life formed there, it did so via plasma (Not to mention isn't plasma+blood some kind of life giving reaction? I forget)
* These skeletons are much weaker than regular skeletons, They need to breathe, they're not immune to viruses, not cold immune, not heat immune etc.

Need I go on?

**Comment by WJohn** [comment]
We have plantmen, shadowmen, golem people, adamantine people and so on. These are all very rare too, but there's no need to throw them out.

SS13 is absolutely FULL of gimmicks, pudl. What's wrong with adding another one?

**Comment by SconesC** [comment]
"how does any organic life form survive without oxygen"

lol

**Comment by WJohn** [comment]
Also as an additional stretch goal for this PR, since you're touching on being able to be set on fire and made the original "people can spread fire onto others" PR, could you make it so fire eventually stops burning on a corpse after a while? Right now you can catch fire from corpses that have been on the floor and on fire for the past half an hour, and it's really annoying.

**Comment by RemieRichards** [comment]
@WJohn not really in scope.
Fire was only touched (visually) for plasmamen, it just evolved into a visual overhaul of fire.

**Comment by Fayrik** [comment]
Can I just make a silly "MUH EMMULSIONS" request here? Okay... So.
I think breathing plasma should actually generate CO2, not O2.
Why? Because that's what a toxins burn mix outputs, so all likelihood is, the chemical reactions are the same, if not similar.

...Also breathing out O2 means they'd gradually become more and more flammable in their own atmosphere, which is silly amounts of dangerous.
(In a well adjusted atmosphere, these guys should probably be able to smoke cigars and pipes.)

**Comment by RemieRichards** [comment]
@Fayrik oh that's true, I didn't even think that their own breathing would cause them to ignite, but it totally would.

CO2 makes more sense then.

**Comment by Kromgar** [comment]
Silicon based lifeforms would  breathe quartz i shit you not

**Comment by GoonOnMoon** [comment]
I think I'm going to add a Where's Waldo gimmick to irritate pudl.

**Comment by Fayrik** [comment]
Awesome, this feels a whole lot better now that they're not just going to kill themselves by breathing.

I must say, I wonder what happens when they're on unsimulated turf. Is there just an atmosphere that doesn't get changed, or does the breath check get skipped?

**Comment by as334** [comment]
I like this, but my only concern is giving C02 a warnining sign, it removes it's use as a way to stealthily choke people using atmos.

**Comment by RemieRichards** [comment]
@Fayrik I assume they don't breathe on those, since the gases their are "unsimulated"

**Comment by SconesC** [comment]
I have no idea what these will be used for but I'm pretty excited nonetheless

**Comment by RemieRichards** [comment]
~~fun~~

**Comment by Fayrik** [comment]
When this is merged, I promise I will choke myself out on an unsimulated tile, just for you, @RemieRichards

**Comment by RemieRichards** [comment]
@Fayrik L-lewd

**Comment by WJohn** [comment]
Oh, I have one last request real quick. I just took a second look at the new on-fire sprites when testing it on humans, and noticed the yellow eyes on the sprite looked pretty ugly and were redundant since the plasma skeletons already had these.

Would you mind replacing those with this version instead?
http://www.mediafire.com/download/d4jf9w8g2tcifni/SKELETONS.rar

**Comment by RemieRichards** [comment]
@WJohn no because I already removed the yellow eyes :P 
I *DO* come from the future after all, I see these things ahead of time :3

**Comment by WJohn** [comment]
Thanks <3

**Pull request merged**